app/src/main/java/com/example/mytestproject/ui/MainActivity.kt

```diff
@@ -1,56 +1,54 @@
 package com.example.mytestproject.ui
 
-import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
-import android.view.MenuItem
-import androidx.appcompat.app.ActionBarDrawerToggle
-import androidx.core.view.GravityCompat
-import androidx.fragment.app.Fragment
+import android.view.View
+import androidx.appcompat.app.AppCompatActivity
+import androidx.navigation.NavController
+import androidx.navigation.findNavController
+import androidx.navigation.ui.AppBarConfiguration
+import androidx.navigation.ui.navigateUp
+import androidx.navigation.ui.setupActionBarWithNavController
+import androidx.navigation.ui.setupWithNavController
 import com.example.mytestproject.R
-import com.example.mytestproject.ui.weatherData.WeatherDataFragment
-import com.google.android.material.navigation.NavigationView
 import kotlinx.android.synthetic.main.activity_main.*
 
-class MainActivity : AppCompatActivity(), NavigationView.OnNavigationItemSelectedListener {
+class MainActivity : AppCompatActivity(){
+
+    private lateinit var appBarConfiguration: AppBarConfiguration
+    private lateinit var navController: NavController
 
-    private lateinit var fragment: Fragment
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
 
-        nav_view.setNavigationItemSelectedListener(this)
         setSupportActionBar(toolbar)
-        val toggle = ActionBarDrawerToggle(
-            this, drawer_layout, toolbar,
-            R.string.navigation_drawer_open,
-            R.string.navigation_drawer_close
+
+        navController = findNavController(R.id.nav_host_fragment)
+
+        appBarConfiguration = AppBarConfiguration(
+            setOf(
+                R.id.nav_second_fragment,
+                R.id.nav_third_fragment,
+                R.id.nav_today_weather,
+                R.id.nav_three_days_weather,
+                R.id.nav_ten_days_fragment
+            ), drawer_layout
         )
-        drawer_layout.addDrawerListener(toggle)
-        toggle.syncState()
-    }
 
-    override fun onNavigationItemSelected(item: MenuItem): Boolean {
-        when (item.itemId) {
-            R.id.nav_first_fragment -> fragment =
-                WeatherDataFragment()
-            R.id.nav_second_fragment -> fragment =
-                FragmentSecond()
-            R.id.nav_third_fragment -> fragment =
-                FragmentThird()
+        setupActionBarWithNavController(navController, appBarConfiguration)
+        nav_view.setupWithNavController(navController)
+        bottom_navigation.setupWithNavController(navController)
+        
+        navController.addOnDestinationChangedListener { _, destination, _ ->
+           when (destination.id) {
+                R.id.nav_second_fragment,R.id.nav_third_fragment -> bottom_navigation.visibility = View.GONE
+                else -> bottom_navigation.visibility = View.VISIBLE
+            }
```

Comment:
Could be rewritten in this way:
```suggestion
bottom_navigation.visibility = when (destination.id) {
                R.id.nav_second_fragment,R.id.nav_third_fragment -> View.GONE
                else -> View.VISIBLE
            }
```

Reply:
Fixed